File: go/cars-assemble/cars_assemble.go
package cars

const (
	NumberOfMinutesInHour   = 60
	CarCountForGroupPackage = 10
	PackageCarCost          = 9_5000
	IndividualCarCost       = 10_000
)

// CalculateWorkingCarsPerHour calculates how many working cars are
// produced by the assembly line every hour.
func CalculateWorkingCarsPerHour(productionRate int, successRate float64) float64 {
	return float64(productionRate) * (successRate / float64(100))
}

// CalculateWorkingCarsPerMinute calculates how many working cars are
// produced by the assembly line every minute.
func CalculateWorkingCarsPerMinute(productionRate int, successRate float64) int {
	return int(CalculateWorkingCarsPerHour(productionRate, successRate) / NumberOfMinutesInHour)
}

// CalculateCost works out the cost of producing the given number of cars.
func CalculateCost(carsCount int) uint {
	groupPackageCount := uint(carsCount / CarCountForGroupPackage)
	individualCarCount := uint(carsCount % CarCountForGroupPackage)

	return uint(groupPackageCount)*PackageCarCost + (individualCarCount * IndividualCarCost)
}

File: clojure/cars-assemble/src/cars_assemble.clj
(ns cars-assemble)

(def ^:private CARS_PRODUCED_PER_HOUR 221.0)
(def ^:private MINUTES_PER_HOUR 60)

(defn round-to-decimal-places
  [num places]
  (let [scale (Math/pow 10 places)]
    (/ (Math/round (* num scale)) scale)))

(defn production-rate
  "Returns the assembly line's production rate per hour,
   taking into account its success rate"
  [speed]
  (let [rate 
    (* speed
      (cond 
        (> speed 9) (* CARS_PRODUCED_PER_HOUR 0.77)
        (> speed 8) (* CARS_PRODUCED_PER_HOUR 0.8)
        (> speed 4) (* CARS_PRODUCED_PER_HOUR 0.9)
        (> speed 0) CARS_PRODUCED_PER_HOUR
        :else 0))]
    (round-to-decimal-places rate 1)))

(defn working-items
  "Calculates how many working cars are produced per minute"
  [speed]
  (int (/ (production-rate speed) MINUTES_PER_HOUR)))